Sensus.Shared/SensusServiceHelper.cs

```diff
@@ -1462,6 +1462,29 @@ private void RemoveScripts(bool issueNotification, params Script[] scripts)
                 IssuePendingSurveysNotificationAsync(null, false);
             }
         }
+
+        //A quick binary search to determine the index placement in order to insert a new Script
+        private int ScriptToAddIndex(Script item)
+        {
+            List<Script> scripts = _scriptsToRun.ToList();

```

Comment:
Is the following code better overall than adding `item` to `scripts`, sorting `scripts`, and returning the index of `item`? This would be three lines of easy-to-read code. Not just a rhetorical question...let me know if you think the code below is a better approach.

Reply:
So, that was the original approach that I was going to take.  But after thinking about it further, since the list will always be in order so there is no need for sorting the list after each insert -- just finding the index.  The list class offers a binary search so I took advantage of that.  Then i started thinking about the general data that will be coming in.  My hypothesis is that a majority of the surveys that come in will either be the first first survey or will need to go at the end of the list.  So I added the scripts.Count == 0 and checked the last item before doing the binary search.  This could be simplified to just lines 1480 - 1482 (returning the index) but the current implementation will be faster and use less computations.  Apologies if this was overthinking -- I have added comments but let me know if you'd like me to simplify the solution.